src/Mails_PT.php

```diff
@@ -154,6 +157,32 @@ public function addMetaboxes() {
 			'id'   => self::POST_TYPE_NAME . '_attachments',
 			'type' => 'file_list',
 		) );
+			
+		//Get Filtered Timber Context
+		$context = apply_filters('juvo_mail_editor_timber_context', Timber::context());
+
+		if (isset($_GET['post'])) {
+
+			//Get This Post's Triggers Slugs
+			$posts_triggers = array_column(get_the_terms($_GET['post'], Mail_Trigger_TAX::TAXONOMY_NAME), 'slug');
+
+			//Array for This Post's Specific Variables 
+			$specific_variables = [];
+
+			//Go Throught All Slugs and get the key which represents the Specific Variable
+			foreach ($posts_triggers as $key => $trigger) {
+				$trigger_specific_variable = key(apply_filters("juvo_mail_editor_{$trigger}_placeholders", array(), array()));
+				if (!empty($trigger_specific_variable) && !in_array($trigger_specific_variable, $specific_variables)) {
+					$specific_variables[$trigger_specific_variable] = $trigger_specific_variable;
+				}
+			}
+
+			//Add The Specific Variables to Timber::context
+			$context["Specific Variables"] = $specific_variables;
+		}
+
+		//Call the function creating the new Variables Field
+		$this->custom_variables_field($cmb, $context);
```

Comment:
The code looks great :). Please do not mix the declarative code of boxes with logic. Move this code into a function, maybe called "get_placeholders_for_current_post". Please use exit early instead of wrapping the whole code in an if block. Make the code always return an array and set array as return type.